File: JavaLab/src/main/java/ch3/section4_static/Quiz1.java
package ch3.section4_static;

public class Quiz1 {
    int a = 0;
    {
        System.out.println("1");
    }
    static {
        System.out.println("2");
    }
    Quiz1(){
        System.out.println("3");
    }
    Quiz1(int no){
        this();
        System.out.println("4");
    }

    public static void main(String[] args) {
        new Quiz1();
        new Quiz1(10);
    }
}

//2 -> 1 -> 3 -> 1 -> 3 -> 4

File: KotlinLab/KotlinLab/src/main/kotlin/ch3/section1_constructor/Quiz1.kt
package org.example.ch3.section1_constructor.quiz1

class User(var name: String?, var age: Int = 0){
    constructor(): this(null, 0)
    constructor(name: String): this(name, 0)
    fun printUserInfo(){
        println("name : $name, age :  $age")
    }
}

fun main() {
    val user1 = User()
    user1.printUserInfo()

    val user2 = User("kim")
    user2.printUserInfo()

    val user3 = User("kim", 30)
    user3.printUserInfo()
}